operator/pkg/ciliumendpointbatch/manager.go

```diff
@@ -0,0 +1,337 @@
+//    create CEB, by passing unique name.
+func (c *cebManagerFcfs) createCeb(name string) *ceBatch {
+	var cebName string = name
+	if name == "" {
+		cebName = uniqueCeBatchName(c.cacheCeb)
+	}
+	ceb := &ceBatch{
+		ceb: &cilium_v2.CiliumEndpointBatch{
+			TypeMeta: meta_v1.TypeMeta{
+				Kind:       "CiliumEndpointBatch",
+				APIVersion: cilium_v2.SchemeGroupVersion.String(),
+			},
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name: cebName,
+				Annotations: map[string]string{
+					annotation.CiliumEndpointBatchQueueInfo: CEBatchingModeFcfs,
+				},
+			},
+			Endpoints: make([]cilium_v2.CoreCiliumEndpoint, 0, c.maxCepsInCeb),
+		},
+	}
+	c.cacheCeb[cebName] = ceb
+	log.Debugf("Generated cebName:%s", cebName)
+	return ceb
+}
+
+// If available, remove Ceb object from cache. deleteCebFromCache called after successful removal from
+// apiserver.
+func (c *cebManagerFcfs) deleteCebFromCache(cebName string) {
+	if _, ok := c.cacheCeb[cebName]; !ok {
+		log.Debugf("Failed to retrieve Ceb object in local cache. cebName:%s", cebName)
+		return
+	}
+	delete(c.cacheCeb, cebName)
+}
+
+// updateCebInCache function copies the ciliumEndpoint object in local cache. if isDeepCopy flag is set,
+// whole CoreCiliumEndpoint object stored in local cache.
+// There are two scenarios updateCebInCache is called.
+// 1) During operator warm boot[after crash or software upgrade], CEB controller sync CEB states from
+// api-server to cache. In this case, isDeepCopy set to true to copy entire CEP object locally.
+// 2) During runtime, reconciler sync curremt state with API server and update meta header only.
+// isDeepCopy flag is set to false.
+func (c *cebManagerFcfs) updateCebInCache(srcCeb *cilium_v2.CiliumEndpointBatch, isDeepCopy bool) {
+	if ceb, ok := c.cacheCeb[srcCeb.GetName()]; ok {
+		if !isDeepCopy {
+			ceb.ceb.ObjectMeta = srcCeb.ObjectMeta
+		} else {
+			srcCeb.DeepCopyInto(ceb.ceb)
+			for _, cep := range ceb.ceb.Endpoints {
+				// Update the cacheCepMap, to reflect all CEPs are packed in a CEB
+				c.cacheCepMap.insert(getCepNameFromCCEP(&cep), srcCeb.GetName())
+			}
+		}
+	}
+}
+
+// If available, getCebFromCache returns CiliumEndpointBatch object.
+func (c *cebManagerFcfs) getCebFromCache(cebName string) (*cilium_v2.CiliumEndpointBatch, error) {
+	if ceb, ok := c.cacheCeb[cebName]; ok {
+		outCeb := new(cilium_v2.CiliumEndpointBatch)
+		ceb.backendMutex.Lock()
+		ceb.ceb.DeepCopyInto(outCeb)
+		ceb.backendMutex.Unlock()
+		return outCeb, nil
+
+	}
+	return nil, fmt.Errorf("Failed to get CEB from local cache for the cebName: %s", cebName)
+}
+
+// findCeb returns the available ceBatch object for a CEP, CEB is chosen based on FCFS.
+// if all CEB's reached max capacity or marked for delete. Allocate a new CEB.
+// if new ceBatch is created, it returns ceBatch object and true
+// else return existing ceBatch and false.
+func (c *cebManagerFcfs) findCeb(cep *cilium_v2.CoreCiliumEndpoint) (*ceBatch, bool) {
```

Comment:
`cep` is not used in the function.

Reply:
In this `fcfs` algorithm we don't look at `cep`. I have another algorithm, that queues CEPs in a CEB based on Identity value. That requires CEP.

Reply:
But this is managerFcfs, I assume identity batching will be another instantiation, so either remove the unused param here or name this manager more generally and leave a comment.

Reply:
>>I assume identity batching will be another instantiation
The signature of interface functions remains same across multiple algorithm. Anyways i removed `cep` for now, because other algorithms is not yet in.